api/query_handler.go

```diff
@@ -85,7 +88,14 @@ func (handler *QueryHandler) HandleAQL(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	handler.handleAQLInternal(aqlRequest, w, r)
+	available := handler.workerPool.GoIfAvailable(func() {
```

Comment:
for query, better use some timeout also

Reply:
no need to do timeout in such case, this is the same behavior as what we are doing currently on limit serve, only difference is separate query and ingestion